src/Params/ADnoteParameters.cpp

```diff
@@ -1516,36 +1516,41 @@ void ADnoteVoiceParam::getfromXML(XMLwrapper& xml, unsigned nvoice)
 void ADnoteParameters::requestWavetables(rtosc::ThreadLink* bToU, int part, int kit)
 {
     for (std::size_t v = 0; v < NUM_VOICES; ++v)
+    {
         VoicePar[v].requestWavetables(bToU, part, kit, v);
+    }
 }
 
 void ADnoteVoiceParam::requestWavetables(rtosc::ThreadLink* bToU, int part, int kit, int voice)
 {
     const bool notFmAndFm[] = { false, true };
-    for(bool isFmSmp : notFmAndFm)
+    if(Enabled)
```

Comment:
Just talking through what I'm reading. So, when a user enables a patch, then next buffer processed after that enable will trigger this block of code which will then step into that particular voice. The voice in question will need to be updated and will immediately spawn a request to populate its wavetable.

Seems right to me.

Reply:
Good point. Basically, that's the idea.

There seems to be a small window: When the user enables a voice, and quickly enough (e.g. by a DAW) a note gets played before Master gets its waves, then something like silence/default sine/undefined would be played (I need to check). At least we must make sure it does not segfault...

But what's worse is that the scale tensors would not be requested right now. This needs to be fixed IMO.